Temp table docs

ericharmeling opened the pull request:
Fixes #6990. 
Fixes #6989.
Fixes #5959.
Fixes #6444.
Fixes #6925.

- Updated `CREATE TABLE` and `CREATE TABLE AS` diagrams and syntax parameters.
- Updated experimental features page.
- Added new page on Temporary Tables with description of temp tables and usage examples.
- Added new diagram on temp tables.
- Updated `CREATE VIEW` diagram and syntax parameters.
- Updated example output for `CREATE VIEW` examples.
- Added new section on Views doc page about temp views.
- Updated `CREATE SEQUENCE` diagram and syntax parameters.
- Updated example output for `CREATE SEQUENCES` examples.
- Added new section in `CREATE SEQUENCE` page about temp sequences.
- Added note about background cleanup job to Temporary Tables page.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/6862)
<!-- Reviewable:end -->

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/3b5a12b49de0f2ebd8cd5b44aa13c1aeaccb6193/

Edited pages:
- [v20.1/create-table-as.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/3b5a12b49de0f2ebd8cd5b44aa13c1aeaccb6193/v20.1/create-table-as.html)
- [v20.1/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/3b5a12b49de0f2ebd8cd5b44aa13c1aeaccb6193/v20.1/create-table.html)
- [v20.1/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/3b5a12b49de0f2ebd8cd5b44aa13c1aeaccb6193/v20.1/experimental-features.html)
- [v20.1/temp-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/3b5a12b49de0f2ebd8cd5b44aa13c1aeaccb6193/v20.1/temp-tables.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/7daa26b9ecb5f087076c54cb4a14a77d53b72fb7/

Edited pages:
- [v20.1/create-table-as.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/7daa26b9ecb5f087076c54cb4a14a77d53b72fb7/v20.1/create-table-as.html)
- [v20.1/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/7daa26b9ecb5f087076c54cb4a14a77d53b72fb7/v20.1/create-table.html)
- [v20.1/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/7daa26b9ecb5f087076c54cb4a14a77d53b72fb7/v20.1/experimental-features.html)
- [v20.1/temp-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/7daa26b9ecb5f087076c54cb4a14a77d53b72fb7/v20.1/temp-tables.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/86edce4d021404dd42bfb645e37d07f46c9b61b8/

Edited pages:
- [v20.1/create-table-as.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/86edce4d021404dd42bfb645e37d07f46c9b61b8/v20.1/create-table-as.html)
- [v20.1/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/86edce4d021404dd42bfb645e37d07f46c9b61b8/v20.1/create-table.html)
- [v20.1/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/86edce4d021404dd42bfb645e37d07f46c9b61b8/v20.1/experimental-features.html)
- [v20.1/temp-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/86edce4d021404dd42bfb645e37d07f46c9b61b8/v20.1/temp-tables.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1cb2fe4e164cda7d0c641ebd9823a62f564ca6c/

Edited pages:
- [v20.1/create-table-as.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1cb2fe4e164cda7d0c641ebd9823a62f564ca6c/v20.1/create-table-as.html)
- [v20.1/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1cb2fe4e164cda7d0c641ebd9823a62f564ca6c/v20.1/create-table.html)
- [v20.1/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1cb2fe4e164cda7d0c641ebd9823a62f564ca6c/v20.1/experimental-features.html)
- [v20.1/temp-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1cb2fe4e164cda7d0c641ebd9823a62f564ca6c/v20.1/temp-tables.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/89450a5949b8ee90eb1a42b383e6f6882e3982da/

Edited pages:
- [v20.1/create-sequence.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/89450a5949b8ee90eb1a42b383e6f6882e3982da/v20.1/create-sequence.html)
- [v20.1/create-table-as.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/89450a5949b8ee90eb1a42b383e6f6882e3982da/v20.1/create-table-as.html)
- [v20.1/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/89450a5949b8ee90eb1a42b383e6f6882e3982da/v20.1/create-table.html)
- [v20.1/create-view.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/89450a5949b8ee90eb1a42b383e6f6882e3982da/v20.1/create-view.html)
- [v20.1/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/89450a5949b8ee90eb1a42b383e6f6882e3982da/v20.1/experimental-features.html)
- [v20.1/temporary-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/89450a5949b8ee90eb1a42b383e6f6882e3982da/v20.1/temporary-tables.html)
- [v20.1/views.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/89450a5949b8ee90eb1a42b383e6f6882e3982da/v20.1/views.html)

ericharmeling commented:
@otan 

JK about a couple things:
- I updated "temp table" to "temporary table" in more places. You were right.
- I added temporary sequences and temporary views to this PR. Please review those.
- All updates to the PR since you reviewed it are in a separate commit.

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e98ba13aefb105a8f87aab5198354175f9e8ad15/

Edited pages:
- [v20.1/create-sequence.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e98ba13aefb105a8f87aab5198354175f9e8ad15/v20.1/create-sequence.html)
- [v20.1/create-table-as.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e98ba13aefb105a8f87aab5198354175f9e8ad15/v20.1/create-table-as.html)
- [v20.1/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e98ba13aefb105a8f87aab5198354175f9e8ad15/v20.1/create-table.html)
- [v20.1/create-view.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e98ba13aefb105a8f87aab5198354175f9e8ad15/v20.1/create-view.html)
- [v20.1/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e98ba13aefb105a8f87aab5198354175f9e8ad15/v20.1/experimental-features.html)
- [v20.1/temporary-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e98ba13aefb105a8f87aab5198354175f9e8ad15/v20.1/temporary-tables.html)
- [v20.1/views.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e98ba13aefb105a8f87aab5198354175f9e8ad15/v20.1/views.html)

rmloveland commented on v20.1/create-table-as.md:
```diff
@@ -78,6 +78,7 @@ table td:first-child {
  `family_def` | An optional [column family definition](column-families.html). Column family names must be unique within the table but can have the same name as columns, constraints, or indexes.
  `create_as_constraint_def` | An optional [primary key constraint](primary-key.html).
  `select_stmt` | A [selection query](selection-queries.html) to provide the data.
+ `opt_temp_create_table` | Defines the table as a session-scoped temporary table. For more information, see [Temporary Tables](temporary-tables.html).<br>**Support for temporary tables is experimental**. The interface and output are subject to change.
```
Does this need the "new in v20.1" tag?

rmloveland commented on v20.1/create-table.md:
```diff
@@ -82,6 +88,7 @@ Parameter | Description
 `table_constraint` | An optional, comma-separated list of [table-level constraints](constraints.html). Constraint names must be unique within the table but can have the same name as columns, column families, or indexes.
 `opt_interleave` | You can potentially optimize query performance by [interleaving tables](interleave-in-parent.html), which changes how CockroachDB stores your data.
 `opt_partition_by` | An [enterprise-only](enterprise-licensing.html) option that lets you define table partitions at the row level. You can define table partitions by list or by range. See [Define Table Partitions](partitioning.html) for more information.
+`opt_temp_create_table` | Defines the table as a session-scoped temporary table. For more information, see [Temporary Tables](temporary-tables.html).<br><br>**Support for temporary tables is [experimental](experimental-features.html#temporary-objects)**.
```
Does this need the "new in v20.1" tag?

rmloveland commented on v20.1/create-table.md:
```diff
@@ -29,6 +29,12 @@ The user must have the `CREATE` [privilege](authorization.html#assign-privileges
   {% include {{ page.version.version }}/sql/diagrams/create_table.html %}
 </div>
 
+**opt_temp_create_table ::=**
```
slick include

rmloveland commented on v20.1/create-table.md:
```diff
@@ -154,13 +161,8 @@ In this example, we create secondary and inverted indexes during table creation.
-        ext JSONB,
         CONSTRAINT "primary" PRIMARY KEY (city ASC, id ASC),
         INDEX vehicles_auto_index_fk_city_ref_users (city ASC, owner_id ASC),
-        INVERTED INDEX ix_vehicle_ext (ext),
```
The text says "we create secondary and inverted indexes", but this appears to remove the inverted index creation.  Is this intentional?  If so, does the surrounding text need to be edited?

rmloveland commented on v20.1/temporary-tables.md:
```diff
@@ -0,0 +1,211 @@
+- Temp tables are automatically dropped at the end of the session.
+- A temp table can only be accessed from the session in which it was created.
+- Temp tables persist across transactions in the same session.
+- Persistent tables cannot reference temp tables, but temp tables can reference persistent tables.
```
I think this would be easier to read if the clauses were reversed, since every other sentence starts with "temp tables are X".

E.g., "Temp tables can reference persistent tables, but persistent tables cannot reference temp tables"

rmloveland commented on v20.1/temporary-tables.md:
```diff
@@ -0,0 +1,211 @@
+
+## Syntax
+
+To create a temp table, add the following [`TEMP`/`TEMPORARY`](sql-grammar.html#opt_temp_create_table) to the beginning of a [`CREATE TABLE`](create-table.html) or [`CREATE TABLE AS`](create-table-as.html) statement:
```
This was a little confusing for me.  I wonder if it is needed at all on this page? Given that the examples show how to do it, and the CREATE TABLE* pages show the syntax diagrams I might argue it's not necessary, and we could refer the user to the CREATE TABLE* pages if they care about syntax diagrams.

rmloveland commented on v20.1/temporary-tables.md:
```diff
@@ -0,0 +1,211 @@
+
+## Example
+
+To turn on temp tables:
```
General comment: if you make each example a subhead, I can jump directly to an example showing what I care about.  E.g., maybe I am trying to get the names of the temp tables out of `pg_catalog`.

Also then we (e.g. Support) could link directly to each task when helping users.

rmloveland commented on v20.1/temporary-tables.md:
```diff
@@ -0,0 +1,211 @@
+
+- [`CREATE TABLE`](create-table.html)
+- [`CREATE TABLE AS`](create-table-as.html)
+- [`SHOW CREATE TABLE`](show-create-table.html)
```
I might also link to the temp views and temp sequences docs from down here so the user can go to them next if they read to this point.

rmloveland commented on v20.1/temporary-tables.md:
```diff
@@ -0,0 +1,211 @@
+
+{% include copy-clipboard.html %}
+~~~ sql
+> SHOW CREATE TABLE users;
```
Not sure if the SHOW CREATE and its output is really needed here

rmloveland commented on v20.1/temporary-tables.md:
```diff
@@ -0,0 +1,211 @@
+
+{% include copy-clipboard.html %}
+~~~ sql
+> SHOW CREATE TABLE vehicles;
```
Perhaps the SHOW CREATE is to prove that the index is pointing at the other table?

rmloveland commented on v20.1/temporary-tables.md:
```diff
@@ -0,0 +1,211 @@
+
+You can also use the full name of the temporary schema in the `SHOW` statement (e.g., `SHOW TABLES FROM pg_temp_1584540651942455000_1`).
+
+{{site.data.alerts.callout_info}}
```
This feels slightly less example-y and more about-y, but maybe it's a subhead called "Visibility of temporary tables" or similar?

rmloveland commented on v20.1/views.md:
```diff
@@ -365,6 +339,61 @@ To remove a view, use the [`DROP VIEW`](drop-view.html) statement:
+
+### Usage
+
+To create a temporary view, add [`TEMP`/`TEMPORARY`](sql-grammar.html#opt_temp) clause to the beginning of a [`CREATE VIEW`](create-view.html) statement.
```
I think this mention of the grammar clause is a bit confusing. It's arguably not needed given the example below.

rmloveland commented:
@ericharmeling late comment.  I just saw while working on something else that temp tables are mentioned on [Postgres compatibility](http://cockroachlabs.com/docs/v20.1/postgresql-compatibility.html) page.  Might be worth adding a link from there, or removing the text that says they are not supported (IDK).

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/332b840abee765e7e6a8a7a01c8d0479fb5b8f91/

Edited pages:
- [v20.1/create-sequence.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/332b840abee765e7e6a8a7a01c8d0479fb5b8f91/v20.1/create-sequence.html)
- [v20.1/create-table-as.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/332b840abee765e7e6a8a7a01c8d0479fb5b8f91/v20.1/create-table-as.html)
- [v20.1/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/332b840abee765e7e6a8a7a01c8d0479fb5b8f91/v20.1/create-table.html)
- [v20.1/create-view.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/332b840abee765e7e6a8a7a01c8d0479fb5b8f91/v20.1/create-view.html)
- [v20.1/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/332b840abee765e7e6a8a7a01c8d0479fb5b8f91/v20.1/experimental-features.html)
- [v20.1/temporary-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/332b840abee765e7e6a8a7a01c8d0479fb5b8f91/v20.1/temporary-tables.html)
- [v20.1/views.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/332b840abee765e7e6a8a7a01c8d0479fb5b8f91/v20.1/views.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/53d5b8fc839fa2249d4d0970c9b01cb1312c7b43/

Edited pages:
- [v20.1/create-sequence.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/53d5b8fc839fa2249d4d0970c9b01cb1312c7b43/v20.1/create-sequence.html)
- [v20.1/create-table-as.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/53d5b8fc839fa2249d4d0970c9b01cb1312c7b43/v20.1/create-table-as.html)
- [v20.1/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/53d5b8fc839fa2249d4d0970c9b01cb1312c7b43/v20.1/create-table.html)
- [v20.1/create-view.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/53d5b8fc839fa2249d4d0970c9b01cb1312c7b43/v20.1/create-view.html)
- [v20.1/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/53d5b8fc839fa2249d4d0970c9b01cb1312c7b43/v20.1/experimental-features.html)
- [v20.1/temporary-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/53d5b8fc839fa2249d4d0970c9b01cb1312c7b43/v20.1/temporary-tables.html)
- [v20.1/views.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/53d5b8fc839fa2249d4d0970c9b01cb1312c7b43/v20.1/views.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f461eeff9fa15d06bab423d5b851530476cc2aef/

Edited pages:
- [v20.1/create-sequence.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f461eeff9fa15d06bab423d5b851530476cc2aef/v20.1/create-sequence.html)
- [v20.1/create-table-as.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f461eeff9fa15d06bab423d5b851530476cc2aef/v20.1/create-table-as.html)
- [v20.1/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f461eeff9fa15d06bab423d5b851530476cc2aef/v20.1/create-table.html)
- [v20.1/create-view.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f461eeff9fa15d06bab423d5b851530476cc2aef/v20.1/create-view.html)
- [v20.1/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f461eeff9fa15d06bab423d5b851530476cc2aef/v20.1/experimental-features.html)
- [v20.1/temporary-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f461eeff9fa15d06bab423d5b851530476cc2aef/v20.1/temporary-tables.html)
- [v20.1/views.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f461eeff9fa15d06bab423d5b851530476cc2aef/v20.1/views.html)

rmloveland commented:
> can you look over that last two commits to make sure everything is in order?

Everything looked good to me.

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/36b473c305792ede4f642551090e519ce6738221/

Edited pages:
- [v20.1/create-sequence.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/36b473c305792ede4f642551090e519ce6738221/v20.1/create-sequence.html)
- [v20.1/create-table-as.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/36b473c305792ede4f642551090e519ce6738221/v20.1/create-table-as.html)
- [v20.1/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/36b473c305792ede4f642551090e519ce6738221/v20.1/create-table.html)
- [v20.1/create-view.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/36b473c305792ede4f642551090e519ce6738221/v20.1/create-view.html)
- [v20.1/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/36b473c305792ede4f642551090e519ce6738221/v20.1/experimental-features.html)
- [v20.1/temporary-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/36b473c305792ede4f642551090e519ce6738221/v20.1/temporary-tables.html)
- [v20.1/views.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/36b473c305792ede4f642551090e519ce6738221/v20.1/views.html)
